src/Overlay/GlobalOverlay.tsx

```diff
@@ -20,24 +20,12 @@ export const GlobalOverlay: FC<GlobalOverlayProps> = ({
     overlayIndex,
     portalIndex,
     setOpen,
-    isOpen
-  } = useGlobalOverlay(rest);
-  const mounted = useRef<boolean>(false);
+  } = useGlobalOverlay({ onClose, ...rest });
   const renderFn = children || render;
 
   useEffect(() => {
     setOpen(open);
   }, [open]);
 
-  useEffect(() => {
```

Comment:
Why was this removed?

Reply:
I don't think it's necessary since `onClose` is being called within `useGlobalOverlay`. This was causing `onClose` to be called twice